src/main/java/gift/doamin/wishlist/service/WishListService.java

```diff
@@ -1,69 +1,88 @@
 package gift.doamin.wishlist.service;
 
-import gift.doamin.wishlist.dto.WishListForm;
-import gift.doamin.wishlist.entity.WishList;
-import gift.doamin.wishlist.repository.WishListRepository;
+import gift.doamin.product.entity.Product;
+import gift.doamin.product.exception.ProductNotFoundException;
+import gift.doamin.product.repository.JpaProductRepository;
+import gift.doamin.user.entity.User;
+import gift.doamin.user.exception.UserNotFoundException;
+import gift.doamin.user.repository.JpaUserRepository;
+import gift.doamin.wishlist.dto.WishForm;
+import gift.doamin.wishlist.dto.WishParam;
+import gift.doamin.wishlist.entity.Wish;
+import gift.doamin.wishlist.exception.InvalidWishFormException;
+import gift.doamin.wishlist.exception.WishListNotFoundException;
+import gift.doamin.wishlist.repository.JpaWishListRepository;
 import java.util.List;
-import org.springframework.http.HttpStatus;
+import java.util.Optional;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
 @Service
 public class WishListService {
 
-    private final WishListRepository wishListRepository;
+    private final JpaWishListRepository wishListRepository;
+    private final JpaUserRepository jpaUserRepository;
+    private final JpaProductRepository productRepository;
+    private final JpaProductRepository jpaProductRepository;
```

Comment:
더 좋은 방법 없을까요?
wishReposiry 이외에는 필요한지 잘 모르겠습니다.

Reply:
```suggestion
    private final WishListRepository wishListRepository;
```
최대한 이 상태를 유지하시면 좋겠습니다.
일단 JpaWishListRepository에 `@Repository`를 넣으시면 됩니다.
interface를 사용하는 이유입니다. SOLID 중 DIP를 사용한거죠.

Reply:
우선 실수로 두 개나 만들어버린 productRepository중 한 개는 삭제했습니다.
JpaWishListRepository를 제외한 다른 repository를 제거하려면 repository 대신 각각의  Service를 의존하거나 연관관계를 끊고 이전처럼 pk값만 저장하는 방법밖에는 생각나지 않는데 혹시 다른 방법이 있을까요?